issue/src/tui/components.rs

```diff
@@ -192,3 +208,105 @@ impl Component<Message, NoUserEvent> for IssueList {
         }
     }
 }
+
+pub struct CommentList<R> {
+    props: Props,
+    comments: ListState<Comment<R>>,
+}
```

Comment:
```rust
// before
SpecificListComponent<R> {
   state: SharedState<Specific<R>>
}
// after
GenericList<ChildComponent> {
   children: Vec<ChildComponent>,
   state: SpecificState
}
```

Reply:
I'm not sure this will work since every component must provide a component-specific implementation of `Component` in order to receive events: https://github.com/radicle-dev/radicle-cli/pull/172/files#diff-dcbc1cad867aa6da05adc9efaa047bda13bfcdf930b86e9b17bcb8544163eb1bR365. 

They usually call `self.perform()` which would need to generate component-specific command results like "index changed in issue list". Might need to experiment with a generic implementation and see if that would fit in.